src/state/farms/fetchFarmsWithAuctions.ts

```diff
@@ -0,0 +1,59 @@
+import { FARM_AUCTION_HOSTING_IN_DAYS } from 'config/constants'
+import farmAuctionAbi from 'config/abi/farmAuction.json'
+import { getFarmAuctionContract } from 'utils/contractHelpers'
+import { multicallv2 } from 'utils/multicall'
+import { ethersToBigNumber } from 'utils/bigNumber'
+import { BSC_BLOCK_TIME } from 'config'
+import { add } from 'date-fns'
+import { sortAuctionBidders } from '../../views/FarmAuction/helpers'
+
+const fetchFarmsWithAuctions = async (
+  currentBlock: number,
+): Promise<{ winnerFarms: string[]; auctionHostingEndDate: string }> => {
+  const now = Date.now()
+  const farmAuctionContract = getFarmAuctionContract()
+  const currentAuctionId = await farmAuctionContract.currentAuctionId()
+  const [auctionData, [auctionBidders]] = await multicallv2(
+    farmAuctionAbi,
+    [
+      {
+        address: farmAuctionContract.address,
+        name: 'auctions',
+        params: [currentAuctionId],
+      },
+      {
+        address: farmAuctionContract.address,
+        name: 'viewBidsPerAuction',
+        params: [currentAuctionId, 0, 500],
+      },
+    ],
+    { requireSuccess: false },
+  )
+  const blocksUntilBlock = auctionData.endBlock.toNumber() - currentBlock
+  if (blocksUntilBlock < 0) {
+    const secondsUntilStart = blocksUntilBlock * BSC_BLOCK_TIME
+    const currentAuctionEndDate = add(new Date(), { seconds: secondsUntilStart })
+    if (
+      Math.abs(now - add(currentAuctionEndDate, { days: FARM_AUCTION_HOSTING_IN_DAYS }).getTime()) /
```

Comment:
`now` here will no longer be `the now` if `await` functions above are executed slowly.

Reply:
Done